src/Stratis.Features.FederatedPeg/Wallet/TransactionData.cs

```diff
@@ -131,9 +132,29 @@ public bool IsConfirmed()
             return this.BlockHeight != null;
         }
 
+        /// <summary>
+        /// Determines if the transaction is spent and confirmed.
+        /// </summary>
+        /// <returns><c>true</c> if spending details is not null and block height is set on spending details.</returns>
+        [NoTrace]
+        public bool IsSpentAndConfirmed()
+        {
+            if (this.spendingDetails == null)
+                return false;
+
+            return this.spendingDetails.BlockHeight != null;
+        }
+
+        /// <summary>
+        /// Determines whether the transaction is spendable.
+        /// </summary>
+        /// <returns><c>true</c> if the amount is more than the dust threshold and spending details are null.</returns>
         [NoTrace]
         public bool IsSpendable()
         {
+            if (this.Amount < Money.Coins(FederatedPegSettings.DustThreshold))
```

Comment:
maybe rename DustThreshold to smth else to avoid confusion with prev meaning of dust? Smth like UTXOThreshold maybe